### ISSUE TITLE
Support XOAUTH2 for smtp

### ISSUE THREAD

**Issue by therealmik** [kolide/fleet#2053]
It's possible to create oauth scopes for just mail on GSuite, rather than using app-specific-passwords with access to all GSuite features (Drive, etc).

To do this, the XOAUTH2 protocol would need to be supported:
https://developers.google.com/gmail/imap/xoauth2-protocol

This is a relatively simple protocol, and requires only collecting a user & access token - so it could reasonably be squashed into the existing database schema (if you're ok with calling the access token a password).

**Comment by zwass** [comment]
I'd be happy to review a PR for this.

**Comment by therealmik** [comment]
Yep, plan on working on it.  CLA is just being held up by legal.
